File: Assets/src/model/tasks/AttackTask.cs
using System.Collections.Generic;

[System.Serializable]
public class AttackTask : DoOnceTask {
  public AttackTask(Actor actor, Body target) : base(actor) {
    this.target = target;
  }

  public Body target { get; }

  protected override BaseAction GetNextActionImpl() {
    return new AttackBaseAction(actor, target);
  }
}

File: web/src/model/tasks/AttackTask.ts
import { DoOnceTask } from '../ActorTask';
import { AttackBaseAction, type BaseAction } from '../BaseAction';
import type { Actor } from '../Actor';
import type { Body } from '../Body';

export class AttackTask extends DoOnceTask {
  readonly target: Body;

  constructor(actor: Actor, target: Body) {
    super(actor);
    this.target = target;
  }

  protected getNextActionImpl(): BaseAction {
    return new AttackBaseAction(this.actor, this.target);
  }
}
